version: "3.8"

secrets:
  db_passwd:
    #external: true
    file: "/root/.secret/db_passwd"
services:
  gitea:
    image: gitea/gitea:1.14.3
    environment:
      - USER_UID=1000
      - USER_GID=1000
      - "DB_PASSWD_FILE=/run/secrets/db_passwd"
      - DB_TYPE=mysql
      - DB_HOST=172.18.77.102:3306
      - DB_NAME=gitea
      - DB_USER=gitea
    networks:
      - yfgj_net
    volumes:
      - /data/gitea:/data
      - /etc/timezone:/etc/timezone:ro
      - /etc/localtime:/etc/localtime:ro
    ports:
      - "3000:3000"
      - "2224:22"
    networks:
      - "yfgj_net"
    deploy:
      mode: replicated
      replicas: 1
      labels:
        - "traefik.enable=true"
        - "traefik.docker.network=yfgj_net"
        - "traefik.http.services.gitea_gitea.loadbalancer.server.port=3000"
        # http 80
        # - "traefik.http.routers.traefik-ui-80.rule=Host(`traefik.01member.com`)" 
        # - "traefik.http.routers.traefik-ui-80.entrypoints=web"
        # http 8443
        #- "traefik.http.routers.traefik-ui-8443.rule=Host(`traefik-8443.01member.com`)" 
        #- "traefik.http.routers.traefik-ui-8443.entrypoints=webnossl"
        # https 443
        - "traefik.http.middlewares.redirect-https.redirectScheme.scheme=https"
        - "traefik.http.routers.gitea-ui-443.rule=Host(`gitea.ctq6.cn`)"
        - "traefik.http.routers.gitea-ui-443.tls=true"
        - "traefik.http.routers.gitea-ui-443.tls.certresolver=foo"
        - "traefik.http.routers.gitea-ui-443.tls.domains[0].main=ctq6.cn"
        - "traefik.http.routers.gitea-ui-443.tls.domains[0].sans=*.ctq6.cn"
        - "traefik.http.routers.gitea-ui-443.middlewares=redirect-https"
        - "traefik.http.routers.gitea-ui-443.entrypoints=websecure"
      placement:
        constraints: [node.role == manager]
      restart_policy:
        condition: on-failure
        delay: 5s
        max_attempts: 3
        window: 120s
networks:
  yfgj_net:
    external: true
    name: yfgj_net